CHANGELOG.md

```diff
@@ -1,5 +1,12 @@
 # CHANGELOG
 
+## 0.5.12
+
+* Fix bug by deprecating `deployment` argument from exchange downloader `search` command the argument is now optional and ignored.
+* Fix bug by deprecating `deployment` argument from `getVersionByDeployment` the argument is now optional and ignored.
```

Comment:
Is there a specific reason we classified this as a patch fix instead of a major bump? I'm not sure if this can be considered a bug fix as this seems more of an implementation change where we no longer consider the deployed environment, which is breaking.

If we do a major version bump we could just remove the deprecated arguments alongside the warning messages.

Reply:
The only reason to avoid major version bump is to avoid supporting multiple versions. Since the main/only consumers of the raml toolkit is sdks, I think it is fine if we prefer major version bump

Reply:
The raml-toolkit is currently broken, we get the API specs from the latest version deployed to Anypoint Exchange, not the latest API specs. The reasoning for avoiding a new major version is that the update does not introduce new features but what we can consider as addressing an existing bug. The `deployment` argument was broken due to changes in the RAML spec, and the change fixes this by always returning the latest deployed version.

If we choose to do a minor version bump we maintain the API signature, we ensure backward compatibility and provide a smooth transition without forcing code updates in other repositories removing the deployment argument.
It's true that the result changes, but the change can be considered that is correcting the output rather than breaking the API, and we inform users by adding deprecation warnings.  I think makes sense to classify this as a bugfix. But I'm also open to changing the code to a major version bump.